### PR TITLE
Improve the use of checkstyle in the build

### PR DESCRIPTION
**Pull request by gnodet**
This PR improves the build time wrt to checkstyle.
Two other PRs will follow eventually to incorporate https://github.com/codehaus-plexus/modello/pull/116, https://github.com/codehaus-plexus/modello/pull/117 and https://github.com/mojohaus/animal-sniffer/pull/165 if possible.
Those 4 fixes make subsequent builds much smoother.

### CODE REVIEW

**Comment by gnodet** [comment]
The problem with checkstyle is that using the error level as a violation means info and warnings will be ignored. Unfortunately, those are only ignored by the plugin, but checkstyle still see the violations and does not cache the results for files that contains such violations. We need to use info level and some real suppressions so that all files can be cached correctly.

File: .mvn/readme.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+The .mvn folder is needed to be able to use the ${maven.multiModuleProjectDirectory} property.
```
Comment (michael-o) [inline .mvn/readme.txt:1]: directory, not folder

File: maven-core/src/main/java/org/apache/maven/execution/BuildResumptionAnalyzer.java [hunk 2]
```diff
@@ -32,5 +32,5 @@
      * @param result Outcome of the current Maven build.
      * @return A {@link BuildResumptionData} instance or {@link Optional#empty()} if resuming the build is not possible.
      */
-    Optional<BuildResumptionData> determineBuildResumptionData( final MavenExecutionResult result );
+    Optional<BuildResumptionData> determineBuildResumptionData( MavenExecutionResult result );
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/execution/BuildResumptionAnalyzer.java:35]: Why did you remove the `final` keywords?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-core/src/main/java/org/apache/maven/execution/BuildResumptionAnalyzer.java (hunk 2 above)
Comment (gnodet) [reply maven-core/src/main/java/org/apache/maven/execution/BuildResumptionAnalyzer.java:35]: They break recent checkstyle rules, which makes sense because they have no meaning on method declaration afaik.

**Review by michael-o — APPROVED** [review]

**Review by olamy — APPROVED** [review]
